api/types.go

```diff
@@ -191,7 +215,7 @@ type EmbeddingRequest struct {
 
 // EmbeddingResponse is the response from [Client.Embeddings].
 type EmbeddingResponse struct {
-	Embedding []float64 `json:"embedding"`
+	Embedding []float32 `json:"embedding"`
```

Comment:
We can leave the old one as `float64`

Reply:
should be noted that this PR will decrease precision of the old embedding endpoint since it uses server.go Embed, which handles float32s.